frontend/src/metabase/querying/components/FilterValuePicker/FilterValuePicker.unit.spec.tsx

```diff
@@ -518,28 +516,27 @@ describe("StringFilterValuePicker", () => {
       });
 
       userEvent.type(screen.getByLabelText("Filter value"), "a@b.com");
-      expect(screen.getByText("a@b.com")).toBeInTheDocument();
-      expect(onChange).not.toHaveBeenCalled();
+      expect(onChange).toHaveBeenLastCalledWith(["a@b.com"]);
     });
 
-    it("should not allow to create a value when there is the exact match in search results", async () => {
-      const { onChange } = await setupStringPicker({
+    it("should not show free-form input in search results", async () => {
+      await setupStringPicker({
         query,
         stageIndex,
         column,
         values: ["a@b.com"],
         searchValues: {
-          "a@b.com": createMockFieldValues({
+          "a@b": createMockFieldValues({
             field_id: PEOPLE.EMAIL,
             values: [["a@b.com"]],
           }),
         },
       });
 
-      userEvent.type(screen.getByLabelText("Filter value"), "a@b.com");
+      userEvent.type(screen.getByLabelText("Filter value"), "a@b");
       act(() => jest.advanceTimersByTime(1000));
       expect(screen.getByText("a@b.com")).toBeInTheDocument();
-      expect(onChange).not.toHaveBeenCalled();
+      expect(screen.queryByText("a@b")).not.toBeInTheDocument();
```

Comment:
We can/should keep the `expect(onChange).not.toHaveBeenCalled();` assertion

Reply:
Fixed